### PR TITLE
PMM-11687 Add retries for problematic tx

### PR DESCRIPTION
**Pull request by artemgavrilov**
PMM-11687

Build: SUBMODULES-3076

- Links to related pull requests (optional).

### CODE REVIEW

File: managed/services/backup/backup_service.go [hunk 1]
```diff
@@ -84,94 +86,115 @@ func (s *Service) PerformBackup(ctx context.Context, params PerformBackupParams)
...
+		var pgErr *pq.Error
+		if errors.As(errTX, &pgErr) {
+			// Serialization failure error code
+			if pgErr.Code == "40001" {
+				s.l.Infof("Transaction serialization failure, retry iteration %d", i)
+				time.Sleep(time.Duration(rand.Intn(100)*i) * time.Millisecond) //nolint:gosec // jitter
+				continue
+			}
+			s.l.Infof("Unknown pq error, code: %s", pgErr.Code.Name())
 		}
```
Comment (artemgavrilov) [inline managed/services/backup/backup_service.go:194]: That is the idea. Restart transaction on serialization failure.

**Review by BupycHuk — APPROVED** [review]

File: managed/services/backup/backup_service.go [hunk 2]
```diff
@@ -84,94 +86,115 @@ func (s *Service) PerformBackup(ctx context.Context, params PerformBackupParams)
...
-			if artifact, err = models.UpdateArtifact(tx.Querier, artifact.ID, models.UpdateArtifactParams{
-				Status: models.BackupStatusPointer(models.PendingBackupStatus),
-			}); err != nil {
+			if job, dbConfig, err = s.prepareBackupJob(tx.Querier, svc, artifact.ID, jobType, params.Mode, params.DataModel, params.Retries, params.RetryInterval); err != nil { //nolint:lll
 				return err
 			}
+			return nil
+		})
+
+		if errTX == nil || i >= 30 {
```
Comment (atymchuk) [inline managed/services/backup/backup_service.go:181]: Can we document this magic number?

Comment (atymchuk) [inline managed/services/backup/backup_service.go:181]:
```suggestion
        // Cap the number of retries to 30
		if errTX == nil || i >= 30 {
```

Comment (artemgavrilov) [reply managed/services/backup/backup_service.go:182]: In general it will be refactored soon. For now just functioning fix.

p.s. commited your suggestion

**Review by atymchuk — APPROVED** [review]

**Pull request merged**